torch/quantization/observer.py

```diff
@@ -114,7 +114,8 @@ class _ObserverBase(ObserverBase):
     eps: torch.Tensor
 
     def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
-                 reduce_range=False, quant_min=None, quant_max=None):
+                 reduce_range=False, quant_min=None, quant_max=None, factory_kwargs=None) -> None:
```

Comment:
Representative sample 3: Observers already have a `dtype` kwarg, so to modify device / dtype for the buffers they allocate, they take in a `factory_kwargs` kwarg.

The call to `torch.nn.factory_kwargs()` is used to verify that nothing extra is passed in the `factory_kwargs` dict. Note that `torch.nn.factory_kwargs()` now supports `None` as an input, returning an empty dict for this case.

`torch.nn.factory_kwargs()` is no longer used outside of this particular case, as this is the only case with pre-existing `dtype` kwargs.